src/lib/coda_metrics/measured.ml

```diff
@@ -0,0 +1,119 @@
+open Core_kernel
+open Prometheus
+
+module type Metric_info_intf = sig
+  val name : string
+
+  val help : string
+end
+
+module type Hashtbl_limited_intf = sig
+  type _ _real
+
+  type 'a t = private 'a _real
+
+  type key
+
+  val cast : 'a _real -> 'a t
+
+  (* raw *)
+  val data : 'a t -> 'a list
+
+  val mem : _ t -> key -> bool
+
+  val find : 'a t -> key -> 'a option
+
+  val find_exn : 'a t -> key -> 'a
+
+  val iter : 'a t -> f:('a -> unit) -> unit
+
+  val fold : 'a t -> init:'b -> f:(key:key -> data:'a -> 'b -> 'b) -> 'b
+
+  (* wrapped *)
+  val create : unit -> _ t
+
+  val of_alist_exn : (key, 'a) List.Assoc.t -> 'a t
+
+  val remove : _ t -> key -> unit
+
+  val add : 'a t -> key:key -> data:'a -> [`Duplicate | `Ok]
+
+  val add_exn : 'a t -> key:key -> data:'a -> unit
+
+  val set : 'a t -> key:key -> data:'a -> unit
+
+  val change : 'a t -> key -> f:('a option -> 'a option) -> unit
+
+  val update : 'a t -> key -> f:('a option -> 'a) -> unit
+end
+
+module Wrap_singleton_table
+    (Tbl : Hashtbl_intf.S)
+    (Metric_info : Metric_info_intf) :
+  Hashtbl_limited_intf with type key = Tbl.key and type 'a _real = 'a Tbl.t =
+(* we explicitly avoid an equality on the original hashtable type
+     * so that the new interface MUST be used to ensure consistency of
+     * reported metrics *)
+struct
+  include Tbl
```

Comment:
To be sure we’re not missing anything can you do the container pattern thing that Jane street recommends where you do `module T = Tbl` and then manually whitelist the functions you want to reexpose like `let map = T.map`